app/models/manageiq/providers/vmware/infra_manager/vm/reconfigure.rb

```diff
@@ -162,6 +162,25 @@ def disk_mode(dependent, persistent)
     end
   end
 
+  def validate_device_backing(backing)
+    case backing.xsiType
+    when 'VirtualDiskFlatVer2BackingInfo'
```

Comment:
@agrare 
I'm confused with ```VirtualDiskFlatVer2BackingOption``` and ```VirtualDiskFlatVer2BackingInfo```.
Looks like the valid modes defined here is for ```VirtualDiskFlatVer2BackingOption```. 
According to the [ref](https://www.vmware.com/support/developer/vc-sdk/visdk25pubs/ReferenceGuide/vim.vm.device.VirtualDisk.FlatVer2BackingInfo.html), valid modes for ```VirtualDiskFlatVer2BackingInfo``` are:
```
The disk persistence mode. Valid modes are:
persistent
independent_persistent
independent_nonpersistent
nonpersistent
undoable
append
```